lib/devise-security/models/session_limitable.rb

```diff
@@ -31,12 +31,71 @@ def update_unique_session_id!(unique_session_id)
         end
       end
 
+      # Removes old/inactive session if allowed.
+      def allow_limitable_authentication?
+        # Always fallback to default logic when session_traceable is not supported.
+        return true unless devise_modules.include?(:session_traceable)
+
+        opts = session_traceable_condition(active: true)
+        return true if max_active_sessions > session_traceable_adapter.find_all(opts).size
```

Comment:
`session_traceable_adapter` will always be the `session_history_class` ORM compatible on `session_traceable` module. It will count the number of active sessions to check whether it is already on its maximum allowable number of sessions.

Reply:
Unclear to me why we can't just use `SessionHistory`.